src/tools/compiletest/src/load_cfg/itemlist.rs

```diff
@@ -0,0 +1,733 @@
+                    )?
+                    .ok_or_else(|| make_err().unwrap_err())?;
+
+                    *item = Item {
+                        val: ItemVal::RevisionSpecificExpanded { revs, content: Box::new(content) },
+                        pos: item.pos,
+                        used: Cell::new(false),
+                    };
+                }
+                ItemVal::RevisionSpecificExpanded { .. } => unreachable!(
+                    "if this visit has never happened before, there should be \
+                    no expanded nodes."
+                ),
+                _ => (),
+            }
+        }
+
+        Ok(())
+    }
+}
+
+// TODO: parse revisions into `ItemVal`s.
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ItemVal<'src> {
+    /* Flags that can only be set to true */
+    BuildPass,
+    BuildFail,
+    CheckPass,
+    CheckFail,
+    RunPass,
+    RunFail,
+    NoPreferDynamic,
+    NoAutoCheckCfg,
+
+    /* Boolean flags that can be set with `foo-bar` or `no-foo-bar` */
+    ShouldIce(bool),
+    ShouldFail(bool),
+    BuildAuxDocs(bool),
+    ForceHost(bool),
+    CheckStdout(bool),
+    CheckRunResults(bool),
+    DontCheckCompilerStdout(bool),
+    DontCheckCompilerStderr(bool),
+    PrettyExpanded(bool),
+    PrettyCompareOnly(bool),
+    CheckTestLineNumbersMatch(bool),
+    StderrPerBitwidth(bool),
+    Incremental(bool),
+    DontCheckFailureStatus(bool),
+    RunRustfix(bool),
+    RustfixOnlyMachineApplicable(bool),
+    CompareOutputLinesBySubset(bool),
+    KnownBug(bool),
+    RemapSrcBase(bool),
+
+    /* Key-value directivesthat can be set with `foo-bar: baz qux` */
+    ErrorPattern(&'src str),
+    RegexErrorPattern(&'src str),
+    CompileFlags(&'src str),
+    Edition(&'src str),
+    RunFlags(&'src str),
+    PrettyMode(&'src str),
+    AuxBin(&'src str),
+    AuxBuild(&'src str),
+    AuxCrate(&'src str),
+    AuxCodegenBackend(&'src str),
+    ExecEnv(&'src str),
+    UnsetExecEnv(&'src str),
+    RustcEnv(&'src str),
+    UnsetRustcEnv(&'src str),
+    ForbidOutput(&'src str),
+    FailureStatus(&'src str),
+    AssemblyOutput(&'src str),
+    TestMirPass(&'src str),
+    LlvmCovFlags(&'src str),
+    FilecheckFlags(&'src str),
+    /// The revisions list
+    Revisions(&'src str),
+
+    /* Directives that are single keys but have variable suffixes */
+    /// Ignore something specific about this test.
+    ///
+    /// `//@ ignore-x86`
+    Ignore {
+        what: &'src str,
+    },
+    /// Components needed by the test
+    ///
+    /// `//@ needs-rust-lld`
+    Needs {
+        what: &'src str,
+    },
+    /// Run only if conditions are met
+    ///
+    /// `//@ only-linux`
+    Only {
+        what: &'src str,
+    },
+    /// Normalizations make transformations in output files before checking
+    ///
+    /// `//@ normalize-stdout: foo -> bar`
+    /// `//@ normalize-stderr-32bit: bar -> baz`
+    /// `//@ normalize-stderr-test: "h[[:xdigit:]]{16}" -> "h[HASH]"`
```

Comment:
Discussion: I don't recall, does `normalize-*` handle string escapes in regexes correctly?

Reply:
I haven't yet gotten to the reimplementation yet, but I think it just does a regex match. Meaning if there are multiple `->` in the string, results might be surprising.

We should probably figure out some quoting rules and use them everywhere relevant. Do you think it makes sense to do something like shell, `'...'` for literals and `"..."` allowing escapes? (and interpolations, but hopefully not relevant here).